File: hello.js
console.log("Hello");
console.log(typeof("hello"));

console.log((42).toString());
console.log(parseInt("42", 10));

const student = "Brian";
// student = "Arthur"; CANNOT RE-ASSIGN
console.log(student);

let age = 24;
age += 1;
console.log(age);

console.log(student.length);
console.log(student[0]);

console.log(student.slice(2,4)); // slice from index 2 to index 4 excluded
console.log(student.substring(2,4)); // slice from index 2 to index 4 excluded
console.log(student.substring(2)); // slice from index to onwards

console.log(student.toLowerCase());
console.log(student.toUpperCase());

const students = ["Takeshi", "David", "Kanae"]
const joinedStudentsWithComma = students.join() // get a joined string with comma between each value
const joinedStudents = students.join("")
console.log(joinedStudentsWithComma);
console.log(joinedStudents);

console.log(joinedStudentsWithComma.split(","))
console.log(joinedStudentsWithComma.split(""))

// Concatenation
console.log(student + " is amazing");
// Interpolation
console.log(`${student} is amazing`);

File: hello.rb
# puts "Hello"
# p "Hello"
# print "Hello"

# p "hello".class

# p 42.to_s
# p "42".to_i

student = "Brian"
p student.length
p student.size
# p student.count # count is for array

# p student.first # first is for array
p student[0]
p student[2..]
p student[2..4]
p student.slice(2,4) # slice from index 2 to index 4 included
p student.upcase
p student.downcase
p student.capitalize

students = ["Takeshi", "David", "Kanae"]
joined_students = students.join
p joined_students

# Concatenation
p student + " is amazing"
# Interpolation
p "#{student} is amazing"
